client/lib/api/who_service.dart

```diff
@@ -80,3 +86,33 @@ class WhoService {
     return 'WEB';
   }
 }
+
+class _MetricHttpClient extends BaseClient {
+  _MetricHttpClient(this._inner);
+
+  final Client _inner;
+
+  @override
+  Future<StreamedResponse> send(BaseRequest request) async {
+    final metric = FirebasePerformance.instance.newHttpMetric(
+      request.url.toString(),
```

Comment:
There are all sorts of privacy implications of collecting URLs in general, and Firebase Performance collects a variety of data.  Is all of that disclosed in the Privacy Policy?  
Nothing herein is legal advice.

From https://firebase.google.com/support/privacy#performance-monitoring-collected-info:

Firebase Performance Monitoring | Firebase installation IDs, IP addresses | How it helps: Performance Monitoring uses Firebase installation IDs to calculate the number of unique Firebase installations that access network resources, to ensure that access patterns are sufficiently anonymous. It also uses Firebase installation IDs with Firebase Remote Config to manage the rate of performance event reporting. Additionally, it uses IP addresses to map performance events to the countries they originate from. For more information, see Data collection.Retention: Performance Monitoring keeps installation and IP-associated events for 30 days and de-identified performance data for 90 days.
-- | -- | --

and

 Examples of information collected by Performance Monitoring

General device information, such as model, OS, and orientation
RAM and disk size
CPU usage
Carrier (based on Mobile Country and Network Code)
Radio/Network information (for example, WiFi, LTE, 3G)
Country (based on IP address)
Locale/language
App version
App foreground or background state
App package name
Firebase installation IDs
Duration times for automated traces
Network URLs (not including URL parameters or payload content) and the following corresponding information:
Response codes (for example, 403, 200)
Payload size in bytes
Response times

Reply:
FYI - You are also adding Remote Config.

From https://firebase.google.com/support/privacy :

Firebase Remote Config | Firebase installation IDs | How it helps: Remote Config uses Firebase installation IDs to select configuration values to return to end-user devices.Retention: Firebase retains Firebase installation IDs until the Firebase customer makes an API call to delete the ID. After the call, data is removed from live and backup systems within 180 days.
-- | -- | --